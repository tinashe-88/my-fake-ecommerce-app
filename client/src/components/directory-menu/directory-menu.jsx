import React from 'react'
import { connect } from 'react-redux'
import { createStructuredSelector } from 'reselect'
import { selectDirectorySection } from '../../redux/directory/directory.selector'

import MenuItem from '../menu-item/menu-item'


import { DirectoryMenuContainer } from './directory-menu.styles'

const Directory = ({ sections }) => (
  <DirectoryMenuContainer>
    {sections.map(({ id, ...otherProps }) => (
      <MenuItem 
        key={id} 
        {...otherProps}
      />
    ))}
  </DirectoryMenuContainer>
)
const mapStateToProps = createStructuredSelector({
  sections: selectDirectorySection
})
export default connect(mapStateToProps)(Directory)