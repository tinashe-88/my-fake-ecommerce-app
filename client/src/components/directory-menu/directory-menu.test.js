import React from 'react'
import { shallow } from 'enzyme'

import { Directory } from './directory-menu'

it('Should render the Directory component', () => {
  expect(shallow(<Directory sections={[]}/>)).toMatchSnapshot()
})