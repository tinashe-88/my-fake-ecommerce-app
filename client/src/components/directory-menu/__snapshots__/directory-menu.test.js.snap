// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render the Directory component 1`] = `<styled.div />`;
